src/lib/insights/client.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  Hits,
+  InsightsClient,
+  InsightsClientMethod,
+  InsightsClientPayload,
+  Renderer,
+  RenderOptions,
+  Unmounter,
+  WidgetFactory,
+} from '../../types';
+import find from 'lodash/find';
+
+export const inferPayload = (
+  method: InsightsClientMethod,
+  results: RenderOptions['results'],
+  hits: Hits<unknown>,
+  objectIDs: string[]
+): InsightsClientPayload => {
+  const { index, queryID } = results;
+  switch (method) {
+    case 'clickedObjectIDsAfterSearch': {
+      const positions = objectIDs.map(objectID => {
+        const hit = find(hits, hit => hit.objectID === objectID);
+        // NOTE: we might want to optimize this so as we browse though the
+        // hits once, which can get pretty big
+        if (typeof hit === 'undefined') {
+          throw new Error(
+            `Could not find ObjectID ${objectID} in the returned hits.`
+          );
+        }
+        return hit.__position;
+      });
+      return { index, queryID, objectIDs, positions };
+    }
+
+    case 'convertedObjectIDsAfterSearch':
+      return { index, queryID, objectIDs };
+
+    default:
+      throw new Error(`Unsupported method passed ${method}`);
+  }
+};
+
+const _wrapInsightsClient = (
```

Comment:
Not sure about the `_` prefix. We don't use this often in the codebase.

Reply:
It looks pretty common here:
https://github.com/algolia/instantsearch.js/blob/9aa5d9a9fde26f6465276749344922f868da566c/src/lib/InstantSearch.js#L289